core-library/src/main/java/org/silverpeas/core/admin/service/DefaultOrganizationController.java

```diff
@@ -1174,4 +1138,49 @@ public List<SpaceInstLight> getPathToComponent(String componentId) {
     }
     return new ArrayList<>();
   }
+
+  private UserDetail[] removeSensitiveData(final UserDetail[] userDetails) {
+    User requester = User.getCurrentRequester();
+    List<UserDetail> results = removeSensitiveData(List.of(userDetails), requester);
+    return results.toArray(new UserDetail[0]);
+  }
+
+  private <T extends UserDetail> List<T> removeSensitiveData(final List<T> users,
+      final User requester) {
+    if (!users.isEmpty() && requester != null) {
+      return users.stream()
+          .map(u -> removeSensitiveData(u, requester))
+          .collect(Collectors.toList());
+    }
+    return users;
+  }
+
+  private <T extends UserDetail> T removeSensitiveData(T user, final User requester) {
+    if (user != null && requester != null && !requester.getId().equals(user.getId()) &&
+        hasNotDomainAdminRights(requester, user.getDomainId())) {
+      return removeSensitiveDataForNonAdmin(user);
+    }
+    return user;
+  }
+
+  private <T extends UserDetail> T removeSensitiveDataForNonAdmin(T user) {
+    if (user.hasSensitiveData()) {
+      user.setEmailAddress("");
```

Comment:
Adding to my thoughts about my first reading of the PR, there is a problem with a user having sensitive data when it is targeted by a user notification when the targeted user has chosen email as a communication channel.

Reply:
Yes it is done in my current fix (not yet pushed as I am resolving other issues)